test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -263,21 +272,21 @@ database_operation::update_operation(thread_context *tc)
         /* Choose a random key to update. */
         uint64_t key_id =
           random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
-        bool successful_update = tc->update(cursor, coll.id, tc->key_to_string(key_id));
+        bool rollback_required = tc->update(cursor, coll.id, tc->key_to_string(key_id));
 
         /* Reset our cursor to avoid pinning content. */
         testutil_check(cursor->reset(cursor.get()));
 
-        /* We received a rollback in update. */
-        if (!successful_update)
-            continue;
-
         /* Commit the current transaction if we're able to. */
-        tc->transaction.try_commit();
+        if (!rollback_required && tc->transaction.can_commit())
+            rollback_required = tc->transaction.commit();
+
+        if (rollback_required)
+            tc->transaction.rollback();
     }
 
-    /* Make sure the last operation is committed now the work is finished. */
+    /* Make sure the last operation is rolled back now the work is finished. */
     if (tc->transaction.active())
-        tc->transaction.commit();
```

Comment:
Was it an issue to make sure the last operation was committed ?

Reply:
Eh I think its fine to rollback the last operation at least we don't need to check that our last commit needs to be rolled back.

Reply:
Agreed.